src/components/navBar/NavBar.tsx

```diff
@@ -0,0 +1,112 @@
+import styles from "./NavBar.module.scss";
+import { useEffect, useCallback, useState } from "react";
+import { debounce } from "@lib/debounce";
+import FdLogo from "../../icons/FdLogo.svg";
+import Hamburger from "../../icons/Hamburger.svg";
+import Menu from "@components/menu/Menu";
+import { fdmgObject } from "@fdmg/article-xml-json";
+
+let prevScrollpos: number;
+
+interface Props {
+  title: string;
+  contents: fdmgObject[];
+}
+
+export default function NavBar({ title, contents }: Props) {
+  const [showNav, setShowNav] = useState(true);
+  const [progression, setProgression] = useState<number>(0);
+  const [menuVisible, setMenuVisible] = useState(false);
+  if (typeof window !== "undefined") {
+    prevScrollpos = window.pageYOffset;
+  }
```

Comment:
This could probably be replaced with the following `useEffect`. This way you'll be certain that the `window` object is available and this code is executed once the Component gets mounted to the DOM.
```suggestion
  useEffect(() => {
    prevScrollpos = window.pageYOffset;
  }, []);
```